File: 199_Binary_Tree_Right_Side_View.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: TreeNode) -> List[int]:
        result = []
        if root == None:
            return result
        
        queue = [root]
        while queue: 
            size = len(queue) 
            level = []
            for i in range(0, size): 
                c = queue.pop(0)
                level.append(c.val) 
                if c.left != None:
                    queue.append(c.left)
                if c.right != None:
                    queue.append(c.right)
            result.append(level[-1])
        return result 

File: 199_Binary_Tree_Right_Side_View.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> result; 
        if(root == NULL) return result; 
    
        queue<TreeNode*> x; 
        x.push(root); 
        while(!x.empty()){
            vector<int> level; 
            int size = x.size(); 
            for(int i = 0; i < size; i++){
                TreeNode* c = x.front();
                level.push_back(c->val);
                x.pop();
                if(c->left != NULL) x.push(c->left); 
                if(c->right != NULL) x.push(c->right); 
            }
            result.push_back(level[level.size() - 1]); 
        }
        return result; 
    }
};